src/cms/templates/pages/page_tree.html

```diff
@@ -128,6 +157,32 @@ <h2 class="heading font-normal mb-2">{% trans 'Pages' %}</h2>
 
 {% block javascript %}
     <script>
+        function select_all_pages() {
+            var selected_pages = document.getElementsByName("page_selected");
+            if (document.getElementsByName("select_all_pages")[0].checked == true) {
+                selected_pages.forEach(e => e.checked = true);
+            } else {
+                selected_pages.forEach(e => e.checked = false);
+            }
+        }
+
+        function bulk_action_execute() {
+            var action = document.getElementById('bulk_action').value;
+            // match action types
+            if ( action == "archive_pages" ) {
+                // TODO
+            } else { // no previous match, than language code -> XLIFF export
+                var page_selected = document.getElementsByName("page_selected");
+                var pages = [];
+                page_selected.forEach(async function (page_checkbox) {
+                    if(page_checkbox.checked) {
+                        pages = pages.concat(page_checkbox.value);
+                    }
+                });
```

Comment:
```suggestion
                page_selected.forEach(async function (page_checkbox) {
                    if(page_checkbox.checked) {
                        pages = pages.concat(page_checkbox.value);
                    }
                });
                if (pages.length == 0) {
                    return;
                }
```
I just noticed that the bulk action opens a new tab without doing anything when no page is selected... I think not opening a new tab makes more sense in this case (in the future, we could even disable the bulk action button when no page is selected, but for now I think it's sufficient).